azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
I agree with this. DISABLE_Xxx is I think consistent with how Boost's config includes work.

Regexes are a point I brought up early on, GCC until recently, had completely broken std::regex. For the 'Completely Boost free AZMQ' this will necessitate making the minimum compiler version GCC4.9, and I am going to assume 3.5 as the minimum Clang version, unless somebody is really interested in trying it with an earlier version.

Also, I think we should assume the completely Boost free AZMQ is being used with standalone Asio built with ASIO_STANDALONE defined, as our minimum compiler requirements should be sufficient for that, and that the main motivation for this is to be able to not require Boost, at all.